projects/core/src/auth/user-auth/services/oauth-lib-wrapper.service.ts

```diff
@@ -22,15 +23,35 @@ import { AuthConfigService } from './auth-config.service';
 })
 export class OAuthLibWrapperService {
   events$: Observable<OAuthEvent> = this.oAuthService.events;
+  currentBaseSite: BaseSite;
 
   // TODO: Remove platformId dependency in 4.0
   constructor(
     protected oAuthService: OAuthService,
     protected authConfigService: AuthConfigService,
     @Inject(PLATFORM_ID) protected platformId: Object,
-    protected winRef: WindowRef
+    protected winRef: WindowRef,
+    protected baseSiteService?: BaseSiteService
```

Comment:
Could we use inject(BaseSiteService)?

Reply:
Yes we can. But it seems here using optional baseSiteService or inject make no difference?

Reply:
Using inject()  function helps to avoid introducing unnecessary breaking changes,  when adding new dependency to a class doesn't require changing the constructor's signature.

Reply:
see 2.12: https://wiki.one.int.sap/wiki/display/spar/Coding+Guidelines

Reply:
Constructor is already used here, and baseSiteService is optional, it should not be breaking change.

Reply:
but adding parameters for constructor is imcompatbile change?

Reply:
it should not be incompatible as along as it's optional.  Correct me if I'm wrong.

Reply:
adapt to `inject`